--- a/zig/lib/std/Target/xtensa.zig
+++ b/zig/lib/std/Target/xtensa.zig
@@ -14,8 +14,12 @@
     dfpaccel,
     div32,
     exception,
+    expstate,
     extendedl32r,
+    esp32s2ops,
+    esp32s3ops,
     fp,
+    forced_atomics,
     highpriinterrupts,
     highpriinterrupts_level3,
     highpriinterrupts_level4,
@@ -26,6 +30,7 @@
     loop,
     mac16,
     minmax,
+    hifi3,
     miscsr,
     mul16,
     mul32,
@@ -34,6 +39,7 @@
     prid,
     regprotect,
     rvector,
+    s32c1i,
     sext,
     threadptr,
     timers1,
@@ -96,14 +102,34 @@
         .description = "Enable Xtensa Exception option",
         .dependencies = featureSet(&[_]Feature{}),
     };
+    result[@intFromEnum(Feature.expstate)] = .{
+        .llvm_name = "expstate",
+        .description = "Enable Xtensa EXPSTATE option",
+        .dependencies = featureSet(&[_]Feature{}),
+    };
     result[@intFromEnum(Feature.extendedl32r)] = .{
         .llvm_name = "extendedl32r",
         .description = "Enable Xtensa Extended L32R option",
         .dependencies = featureSet(&[_]Feature{}),
     };
+    result[@intFromEnum(Feature.esp32s2ops)] = .{
+        .llvm_name = "esp32s2ops",
+        .description = "Support Xtensa esp32-s2 ISA extension",
+        .dependencies = featureSet(&[_]Feature{}),
+    };
+    result[@intFromEnum(Feature.esp32s3ops)] = .{
+        .llvm_name = "esp32s3ops",
+        .description = "Support Xtensa esp32-s3 ISA extension",
+        .dependencies = featureSet(&[_]Feature{}),
+    };
     result[@intFromEnum(Feature.fp)] = .{
         .llvm_name = "fp",
         .description = "Enable Xtensa Single FP instructions",
+        .dependencies = featureSet(&[_]Feature{}),
+    };
+    result[@intFromEnum(Feature.forced_atomics)] = .{
+        .llvm_name = "forced-atomics",
+        .description = "Assume that lock-free native-width atomics are available",
         .dependencies = featureSet(&[_]Feature{}),
     };
     result[@intFromEnum(Feature.highpriinterrupts)] = .{
@@ -166,6 +192,11 @@
         .description = "Enable Xtensa MINMAX option",
         .dependencies = featureSet(&[_]Feature{}),
     };
+    result[@intFromEnum(Feature.hifi3)] = .{
+        .llvm_name = "hifi3",
+        .description = "Enable Xtensa HIFI3 instructions",
+        .dependencies = featureSet(&[_]Feature{}),
+    };
     result[@intFromEnum(Feature.miscsr)] = .{
         .llvm_name = "miscsr",
         .description = "Enable Xtensa Miscellaneous SR option",
@@ -204,6 +235,11 @@
     result[@intFromEnum(Feature.rvector)] = .{
         .llvm_name = "rvector",
         .description = "Enable Xtensa Relocatable Vector option",
+        .dependencies = featureSet(&[_]Feature{}),
+    };
+    result[@intFromEnum(Feature.s32c1i)] = .{
+        .llvm_name = "s32c1i",
+        .description = "Enable Xtensa S32C1I option",
         .dependencies = featureSet(&[_]Feature{}),
     };
     result[@intFromEnum(Feature.sext)] = .{
@@ -245,6 +281,158 @@
 };
 
 pub const cpu = struct {
+    pub const cnl: CpuModel = .{
+        .name = "cnl",
+        .llvm_name = "cnl",
+        .features = featureSet(&[_]Feature{
+            .bool,
+            .coprocessor,
+            .dcache,
+            .debug,
+            .density,
+            .div32,
+            .exception,
+            .fp,
+            .hifi3,
+            .highpriinterrupts,
+            .interrupt,
+            .loop,
+            .miscsr,
+            .mul32,
+            .mul32high,
+            .nsa,
+            .prid,
+            .regprotect,
+            .rvector,
+            .s32c1i,
+            .sext,
+            .threadptr,
+            .timers1,
+            .windowed,
+        }),
+    };
+    pub const esp32: CpuModel = .{
+        .name = "esp32",
+        .llvm_name = "esp32",
+        .features = featureSet(&[_]Feature{
+            .bool,
+            .clamps,
+            .coprocessor,
+            .dcache,
+            .debug,
+            .density,
+            .dfpaccel,
+            .div32,
+            .exception,
+            .expstate,
+            .fp,
+            .highpriinterrupts,
+            .highpriinterrupts_level7,
+            .interrupt,
+            .loop,
+            .mac16,
+            .minmax,
+            .miscsr,
+            .mul16,
+            .mul32,
+            .mul32high,
+            .nsa,
+            .prid,
+            .regprotect,
+            .rvector,
+            .s32c1i,
+            .sext,
+            .threadptr,
+            .timers3,
+            .windowed,
+        }),
+    };
+    pub const esp32s2: CpuModel = .{
+        .name = "esp32s2",
+        .llvm_name = "esp32s2",
+        .features = featureSet(&[_]Feature{
+            .clamps,
+            .coprocessor,
+            .dcache,
+            .debug,
+            .density,
+            .div32,
+            .esp32s2ops,
+            .exception,
+            .highpriinterrupts,
+            .highpriinterrupts_level7,
+            .interrupt,
+            .minmax,
+            .miscsr,
+            .mul16,
+            .mul32,
+            .mul32high,
+            .nsa,
+            .prid,
+            .regprotect,
+            .rvector,
+            .sext,
+            .threadptr,
+            .timers3,
+            .windowed,
+        }),
+    };
+    pub const esp32s3: CpuModel = .{
+        .name = "esp32s3",
+        .llvm_name = "esp32s3",
+        .features = featureSet(&[_]Feature{
+            .bool,
+            .clamps,
+            .coprocessor,
+            .dcache,
+            .debug,
+            .density,
+            .dfpaccel,
+            .div32,
+            .esp32s3ops,
+            .exception,
+            .fp,
+            .highpriinterrupts,
+            .highpriinterrupts_level7,
+            .interrupt,
+            .loop,
+            .mac16,
+            .minmax,
+            .miscsr,
+            .mul16,
+            .mul32,
+            .mul32high,
+            .nsa,
+            .prid,
+            .regprotect,
+            .rvector,
+            .s32c1i,
+            .sext,
+            .threadptr,
+            .timers3,
+            .windowed,
+        }),
+    };
+    pub const esp8266: CpuModel = .{
+        .name = "esp8266",
+        .llvm_name = "esp8266",
+        .features = featureSet(&[_]Feature{
+            .debug,
+            .density,
+            .exception,
+            .extendedl32r,
+            .highpriinterrupts,
+            .highpriinterrupts_level3,
+            .interrupt,
+            .mul16,
+            .mul32,
+            .nsa,
+            .prid,
+            .regprotect,
+            .rvector,
+            .timers1,
+        }),
+    };
     pub const generic: CpuModel = .{
         .name = "generic",
         .llvm_name = "generic",
