diff --git a/zig/src/target.zig b/zig/src/target.zig
index ee9d39fbf..fe2d1c4e8 100644
--- a/zig/src/target.zig
+++ b/zig/src/target.zig
@@ -200,6 +200,7 @@ pub fn hasLlvmSupport(target: *const std.Target, ofmt: std.Target.ObjectFormat)
         .x86,
         .x86_64,
         .xcore,
+        .xtensa,
         .nvptx,
         .nvptx64,
         .lanai,
@@ -210,7 +211,6 @@ pub fn hasLlvmSupport(target: *const std.Target, ofmt: std.Target.ObjectFormat)
 
         // LLVM backend exists but can produce neither assembly nor object files.
         .csky,
-        .xtensa,
         => false,
 
         // No LLVM backend exists.
