diff --git a/build b/build
index 3d15ab1be..f00aa8a3f 100755
--- a/build
+++ b/build
@@ -55,6 +55,7 @@ cmake "$ROOTDIR/llvm" \
   -DCLANG_TOOL_C_INDEX_TEST_BUILD=OFF \
   -DCLANG_TOOL_ARCMT_TEST_BUILD=OFF \
   -DCLANG_TOOL_C_ARCMT_TEST_BUILD=OFF \
+  -DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=Xtensa \
   -DCLANG_TOOL_LIBCLANG_BUILD=OFF
 cmake --build . --target install
 
@@ -190,6 +191,7 @@ cmake "$ROOTDIR/llvm" \
   -DCLANG_TOOL_ARCMT_TEST_BUILD=OFF \
   -DCLANG_TOOL_C_ARCMT_TEST_BUILD=OFF \
   -DCLANG_TOOL_LIBCLANG_BUILD=OFF \
+  -DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=Xtensa \
   -DLLD_BUILD_TOOLS=OFF
 cmake --build . --target install
 
@@ -198,6 +200,7 @@ $ZIG build \
   --prefix "$ROOTDIR/out/zig-$TARGET-$MCPU" \
   --search-prefix "$ROOTDIR/out/$TARGET-$MCPU" \
   -Dflat \
+  -Dllvm-has-xtensa \
   -Dstatic-llvm \
   -Doptimize=ReleaseFast \
   -Dstrip \
   -Dstrip \

